core/domain/state_domain.py

```diff
@@ -745,16 +748,258 @@ def get_all_html_content_strings(self):
                                 'The solution does not have a valid '
                                 'correct_answer type.')
 
-
-        if self.id in (
-                'ItemSelectionInput', 'MultipleChoiceInput',
-                'DragAndDropSortInput'):
-            customization_args_html_list = (
-                self.customization_args['choices']['value'])
-            html_list = html_list + customization_args_html_list
+        html_list += self.get_all_html_in_cust_args()
 
         return html_list
 
+    @staticmethod
+    def convert_html_in_interaction(interaction_dict, conversion_fn):
+        """Checks for HTML fields in the outcome and converts it
+        according to the conversion function.
+
+        Args:
+            interaction_dict: dict. The interaction dict.
+            conversion_fn: function. The function to be used for converting the
+                HTML.
+
+        Returns:
+            dict. The converted interaction dict.
+        """
+        def convert_cust_args(ca_value, obj_type):
+            """Conversion function that converts html content in
+            customization arguments.
+
+            Args:
+                ca_value: dict|str. The original value of customization
+                    argument.
+                obj_type: str. Indicates the obj_type found in
+                    the customization arguments schema.
+
+            Returns:
+                str. The updated customization argument value.
+            """
+
+            # Because convert_html_in_interaction is used in multiple state
+            # migrations, we must handle the case where the value has not yet
+            # been migrated to a SubtitledHtml object (pre schema v40).
+            if isinstance(
+                    ca_value,
+                    (python_utils.UNICODE, python_utils.BASESTRING)):
+                ca_value = conversion_fn(ca_value)
+            elif obj_type == 'SubtitledHtml':
+                ca_value['html'] = conversion_fn(ca_value['html'])
+
+            return ca_value
+
+        InteractionInstance.convert_content_in_cust_args(
+            interaction_dict['id'],
+            interaction_dict['customization_args'],
+            convert_cust_args)
+        return interaction_dict
+
+    @staticmethod
+    def convert_content_in_cust_args(
```

Comment:
Sorry- when you say you don't see a similar need for Subtitled objects, doesn't SubtitledHtml contain html that we would need to apply transformations? I do agree that it isn't needed for unicode strings, I'll see if I can clean up the code a little bit.

Reply:
I've changed ```convert_content_in_cust_args``` to ```convert_subtitled_content_in_cust_args```. The reason being that the conversion function is passed SubtitledHtml or SubtitledUnicode, rather than the html or unicode. Does that sg?

Reply:
This is no longer relevant, as this function will be removed